### PR TITLE
Francious/variance

### PR DESCRIPTION
**Pull request by RostislavKrotenko**

### CODE REVIEW

**Comment by lhalam** [comment]
Conflicting files
src/main/java/org/codewars/kata/implementation/Francious/EightImpl.java
src/main/java/org/codewars/kata/implementation/Francious/SevenImpl.java

**Review by lhalam — COMMENTED** [review]
точно хоч все це видаляти

File: src/main/java/org/codewars/kata/implementation/Francious/EightImpl.java [hunk 1]
```diff
@@ -4,16 +4,25 @@
 
 import java.util.ArrayList;
 
+
 public class EightImpl implements Eight {
     public int liters(double time) {
-        return (int) Math.floor(time * 0.5);
```
Comment (lhalam) [inline src/main/java/org/codewars/kata/implementation/Francious/EightImpl.java:9]: are you sure you want to delete it?

File: src/main/java/org/codewars/kata/implementation/Francious/EightImpl.java [hunk 2]
```diff
@@ -4,16 +4,25 @@
...
+        return 0;
     }
 
     public double getVolumeOfCuboid(double length, double width, double height) {
-        return length * width * height;
```
Comment (lhalam) [inline src/main/java/org/codewars/kata/implementation/Francious/EightImpl.java:13]: are you sure you want to delete it?

File: src/main/java/org/codewars/kata/implementation/Francious/EightImpl.java [hunk 3]
```diff
@@ -52,10 +61,14 @@ public int[] countPositivesSumNegatives(int[] input) {
     }
 
     public int stringToNumber(String str) {
-        return Integer.parseInt(str);
```
Comment (lhalam) [inline src/main/java/org/codewars/kata/implementation/Francious/EightImpl.java:55]: are you sure you want to delete it?

File: src/main/java/org/codewars/kata/implementation/Francious/FiveImpl.java [hunk 4]
```diff
@@ -35,13 +35,7 @@ public long[] gap(int g, long m, long n) {
     }
 
     public int zeros(int n) {
-        int count = 0;
-
-        while (n > 0) {
-            n /= 5;
-            count += n;
-        }
-        return count;
```
Comment (lhalam) [inline src/main/java/org/codewars/kata/implementation/Francious/FiveImpl.java:44]: are you sure you want to delete it?

File: src/main/java/org/codewars/kata/implementation/Francious/FiveImpl.java [hunk 5]
```diff
@@ -61,57 +55,11 @@ public BigInteger perimeter(BigInteger n) {
...
-            if (Math.abs(result - m) < precision) {
-                return midpoint;
-            } else if (result < m) {
-                lowerBound = midpoint;
-            } else {
-                upperBound = midpoint;
-            }
-        }
-
-        return lowerBound;
```
Comment (lhalam) [inline src/main/java/org/codewars/kata/implementation/Francious/FiveImpl.java:87]: are you sure you want to delete it?

File: src/main/java/org/codewars/kata/implementation/Francious/FiveImpl.java [hunk 6]
```diff
@@ -61,57 +55,11 @@ public BigInteger perimeter(BigInteger n) {
...
-                    minNumStr = currentNumStr;
-                    minIndex = i;
-                    insertIndex = j;
-                }
-                sb.deleteCharAt(j);
-            }
-        }
-
-        long minNum = Long.parseLong(minNumStr);
-        return new long[]{minNum, minIndex, insertIndex};
```
Comment (lhalam) [inline src/main/java/org/codewars/kata/implementation/Francious/FiveImpl.java:114]: are you sure you want to delete it?

File: src/main/java/org/codewars/kata/implementation/Francious/SevenImpl.java [hunk 7]
```diff
@@ -13,11 +13,10 @@ public long newAvg(double[] arr, double navg) {
         double length = arr.length;
         double exptResult = ((length + 1) * navg - result);
 
-        if (exptResult <= 0) {
-            throw new IllegalArgumentException();
-        }
 
-        return (long) Math.ceil(exptResult);
```
Comment (lhalam) [inline src/main/java/org/codewars/kata/implementation/Francious/SevenImpl.java:20]: are you sure you want to delete it?

**Comment by RostislavKrotenko** [comment]
> точно хоч все це видаляти

вже не хочу))

**Pull request merged**